_drafts/gnu-make-guidelines.md

```diff
@@ -0,0 +1,845 @@
+---
+title: "GNU Make Overview and Guidelines"
+description: "A brief overview of GNU make features and suggestions for getting
+the most out of it"
+author: noah
+image: /img/gnu-make-guidelines/gnu-make-guidelines.img
+---
+
+GNU Make is a very commonly used program for building C language software. It's
+used for example when [building the Linux
+kernel](https://wiki.archlinux.org/index.php/Kernel/Traditional_compilation#Compilation),
+and many commonly used GNU/Linux programs and software libraries.
+
+Most embedded software developers will probably run across software that uses
+GNU Make. Though there are [many many
+alternatives](https://en.wikipedia.org/wiki/List_of_build_automation_software),
+it's still commonly used in new and existing software, and it's widely
+supported.
+
+<!-- excerpt start -->
+
+This article explains some general concepts and features of GNU Make, with some
+recommendations for getting the most out of a Make build! Consider it a brief
+guided tour through some of my favorite/most used Make concepts and features 🤗
+
+<!-- excerpt end -->
+
+{:.no_toc}
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+* auto-gen TOC:
+{:toc}
+
+## What is GNU Make?
+
+[GNU Make](https://www.gnu.org/software/make/) is a program that automates
+running shell commands, that helps when running repetitive tasks.
+
+It is typically used to transform files into some other form, eg compiling
+source code into programs or libraries.
+
+It enables tracking "prerequisites" and running a hierarchy of commands to
+produce "targets".
+
+The GNU Make manual is very good:
+[https://www.gnu.org/software/make/manual/html_node/index.html](https://www.gnu.org/software/make/manual/html_node/index.html)
+
+When searching for help on Make, using `gnu make xxx` as the search terms can be
+helpful.
+
+Let's dive in!
+
+## Invoking Make
+
+Running `make` will load a file named `Makefile` from the current directory,
+and attempt to update the default **goal** (more on goals later).
+> *the default is to try `GNUmakefile`, `makefile`, and `Makefile`, in that
+> order*
+
+You can specify another file with the `-f/--file` arg: `make -f foo.mk`
+
+You can specify any number of *goals* by listing them as positional arguments:
+
+```bash
+# typical goals
+make clean all
+```
+
+You can run make in another directory with the `-C` arg:
+
+```bash
+make -C some/sub/directory
+```
+
+Make will run as if it first `cd`'d to that directory.
+
+*Fun fact- `git` also can be run with `-C` for the same effect!*
+
+## Parallel invocation
```

Comment:
I'd move this section further down. Parallel invocation doesn't mean much to a reader if they don't know what it really does yet or how to use it.